File: shopping-list-et.api/shopping-list-et.domain/Entities/ShoppingList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace shopping_list_et.domain.Entities
{
    public class ShoppingList
    {
        public int Id { get; set; }
        public bool Active { get; set; }
        public string? Name { get; set; }
        public int? ItemsCount => Items?.Count ?? 0;
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public List<Item>? Items { get; set; }
    }
}

File: shopping-list-et.app/lib/models/shopping_list.dart
import 'item.dart';

class ShoppingList{
  final int id;
  final bool active;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<Item>? items;

  ShoppingList(this.id, this.active, this.createdAt, this.updatedAt, this.items);

  ShoppingList.fromJson(Map<String, dynamic> json)
          : id = json["id"],
            active = json["active"],
            createdAt = DateTime.parse(json["createdAt"]),
            updatedAt = DateTime.parse(json["updatedAt"]),
            items = json["items"];
}